src/Library/demos/Launcher/main.js

```diff
@@ -0,0 +1,94 @@
+import Gtk from "gi://Gtk";
+import Gdk from "gi://Gdk";
+import Gio from "gi://Gio";
+import GLib from "gi://GLib";
+
+const launch_file = workbench.builder.get_object("launch_file");
+const file_details = workbench.builder.get_object("file_details");
+const file_location = workbench.builder.get_object("file_location");
+const change_file = workbench.builder.get_object("change_file");
+const uri_launch = workbench.builder.get_object("uri_launch");
+const uri_details = workbench.builder.get_object("uri_details");
+
+Gio._promisify(Gtk.FileLauncher.prototype, "launch", "launch_finish");
+Gio._promisify(
+  Gtk.FileLauncher.prototype,
+  "open_containing_folder",
+  "open_containing_folder_finish",
+);
+Gio._promisify(Gtk.FileDialog.prototype, "open", "open_finish");
+Gio._promisify(Gtk.UriLauncher.prototype, "launch", "launch_finish");
+
+const file_launcher = new Gtk.FileLauncher();
+const uri_launcher = new Gtk.UriLauncher();
+const file = Gio.File.new_for_path(pkg.pkgdatadir).resolve_relative_path(
+  "Library/demos/Launcher/workbench.txt",
+);
+file_launcher.set_file(file);
+let text = "";
+
+//File Launcher
+async function launchesFile() {
+  const result = await file_launcher.launch(workbench.window, null);
+  if (result) {
+    console.log("File Launched");
+  }
+}
+
+async function launchesParentFolder() {
+  const result = await file_launcher.open_containing_folder(
+    workbench.window,
+    null,
+  );
+  if (result) {
+    console.log("Parent folder opened");
+  }
+}
+
+async function changesFile() {
```

Comment:
```suggestion
async function changesFile() {
```

Let's do away with these empty wrappers.